raiden-ts/src/utils/ethers.ts

```diff
@@ -43,29 +53,44 @@ export function fromEthersEvent<T>(
       resultSelector,
     ) as Observable<T>;
 
-  let first = 0;
-  let latestEventBlock = 0;
+  const blockQueue: number[] = []; // sorted 'fromBlock' queue, at most of [range] size
   return defer(() => {
-    // 'first' starts with subscription-time's private value set with provider.resetEventsBlock
-    first = (target as any)._lastBlockNumber || target.blockNumber || Number.POSITIVE_INFINITY;
+    // 'resetEventsBlock' is private, set at [[Raiden]] constructor, so we need 'any'
+    const resetBlock: number = (target as any)._lastBlockNumber;
+    const firstBlock = resetBlock && resetBlock > 0 ? resetBlock : target.blockNumber ?? 1;
```

Comment:
Which case would differ `resetBlock && resetBlock > 0` from `resetBlock > 0`? :thinking:

Reply:
The case in whic `resetBlock` is `undefined`. JS allows this comparison (which is always `false`), but TS complains if it discovers we're trying to compare `undefined` with `number`.